src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -87,7 +88,12 @@ impl LogicalScan {
             .unwrap_or_default();
 
         let schema = Schema { fields };
-        let base = PlanBase::new_logical(ctx, schema, pk_indices);
+        let functional_dependency = if !schema.is_empty() {
```

Comment:
This might be incorrect. 

We should check whether `pk_indices` is `None` or `Some` (The result of `.collect::<Option<Vec<_>>>()`). It's `Some` only when **ALL** columns are in the output. When schema is non-empty, pk can still be empty.

Reply:
Also for other nodes.

Reply:
This must be some typo :rofl: I should check the pk_indices here.